test/IRGen/objc_direct.swift

```diff
@@ -0,0 +1,40 @@
+// RUN: %target-swift-frontend -import-objc-header %S/Inputs/objc_direct.h -emit-ir -o - %s | %FileCheck --check-prefix=CHECK-IR %s
```

Comment:
Please also add a SILGen test that uses FileCheck to verify that the call is lowered as a function_ref and not an objc_method.

More ideas for things to test:
- Protocol conformances, as above
- Calling an objc_direct method on AnyObject -- should probably disallow this?
- Can (methods that get imported as) subscripts be objc_direct?

Reply:
I think we also need typechecker tests that we don't allow overriding, either handle or disallow super calls, and so on.  Probably the override part means importing the method as final.

Reply:
Thanks for the suggestions! The tests have been refactored into `SILGen`, `IRGen`, `Interpreter`, and `ClangImporter` (disallow `objc_direct` on `AnyObject`). I've also added subscript methods to the tests.

Reply:
@rjmccall I've added a test in `Interpreter/objc_direct.swift` and it appears that overriding `objc_direct` methods works fine.
https://github.com/apple/swift/pull/33349/files#diff-be1eb3d9c772960853ed4997fcf00754R43

Reply:
What you're testing is whether an imported direct method can be used as a protocol witness.  That's good, but it's not what I meant; I meant a class method override.  You should check that you can't define a subclass of an ObjC class that overrides a direct method.

Reply:
By the way, I've added test for this in `test/type/objc_direct.swift`.